### PR TITLE
algorithms for cographs

### PR DESCRIPTION
**Pull request by jnyrup**
Two algorithms for cographs:
- modular decomposition of cographs into cotrees
- verification of cographs

### CODE REVIEW

**Comment by ysitu** [comment]
Cographs can be recognized in linear time. But it seems that your algorithm is not linear-time since it depends on explicitly forming complement graphs.

File: networkx/algorithms/cographs.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        else:
+            return isCographAux(nx.complement(G))
+    else:
+        raise nx.NetworkXError("Input is not a correct NetworkX graph.")
+
+
+def isCographAux(G):
+    modules = nx.connected_component_subgraphs(nx.complement(G))
+    if len(modules) == 1:
+        if len(modules[0]) == 1:
```
Comment (jfinkels) [inline networkx/algorithms/cographs.py]: Simpler: `return len(modules[0]) == 1`.

File: networkx/algorithms/cographs.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            # return leaf node
+            return True
+        else:
+            return False
+    else:
```
Comment (jfinkels) [inline networkx/algorithms/cographs.py]: Remove this redundant `else`.

File: networkx/algorithms/cographs.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+
+def decomp(G, Gres, rootLabel):
+    # list of modules from complement graph
+    modules = nx.connected_component_subgraphs(nx.complement(G))
+    if len(modules) == 1:
+        if len(modules[0]) == 1:
+            # return leaf node
+            return modules[0].nodes()[0]
+        else:
```
Comment (jfinkels) [inline networkx/algorithms/cographs.py]: Remove redundant `else` line.

File: networkx/algorithms/cographs.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            raise nx.NetworkXUnfeasible(
+                "input graph is not a valid cograph and corresponding cotree cannot be computed")
+    else:
```
Comment (jfinkels) [inline networkx/algorithms/cographs.py]: Remove redundant `else` line.

**Comment by jfinkels** [comment]
Lots of PEP8 code style issues here. Please run your code through a style checker, like [`flake8`][1], for example. Also, naming in Python follows the `separated_by_underscores` convention instead of the `camelCase` convention.

[1]: https://flake8.readthedocs.org/en/2.3.0/

**Comment by MridulS** [comment]
You should also add tests for the same. 
I see you have added some tests in https://github.com/jnyrup/thisandthat/blob/master/cographTest.py . NetworkX uses [nosetests](https://nose.readthedocs.org/en/latest/), please add tests according to it. 
You can view other test files in the codebase for inspiration.

**Comment by ysitu** [comment]
Before jumping into code style changes and adding tests, are we sure that we can include the algorithms as is?

**Comment by MridulS** [comment]
@jnyrup sorry for the super late ping on this one 🙃

Let me know if you still have the bandwidth/interest in getting this into networkx :)

**Comment by jnyrup** [comment]
Long time no see 😄 

I initially opened this PR with some code I hacked together for an assignment in a university course, but with little experience in contributing to OSS.
I haven't used this project since and I have written close to zero lines of Python for the past seven years, so it's no longer something I have an interest in completing.

But feel free to use any of the code in here.

**Comment by MridulS** [comment]
No worries. Thanks for the contributions :)

**Pull request closed without merging**